waddrmgr/manager_test.go

```diff
@@ -1019,38 +1019,66 @@ func testImportScript(tc *testContext) bool {
 
 // testMarkUsed ensures used addresses are flagged as such.
 func testMarkUsed(tc *testContext) bool {
-	expectedAddr1 := expectedAddr{
-		addressHash: hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
-		used:        true,
+	tests := []struct {
+		name       string
+		addrType   btcutil.Address
+		in         []byte
+		blockstamp waddrmgr.BlockStamp
+	}{
+		{
+			name:     "managed address",
+			addrType: &btcutil.AddressPubKeyHash{},
+			in:       hexToBytes("2ef94abb9ee8f785d087c3ec8d6ee467e92d0d0a"),
+		},
+		{
+			name:     "script address",
+			addrType: &btcutil.AddressScriptHash{},
+			in:       hexToBytes("da6e6a632d96dc5530d7b3c9f3017725d023093e"),
+		},
 	}
+
 	prefix := "MarkUsed"
 	chainParams := tc.manager.ChainParams()
-	addrHash := expectedAddr1.addressHash
-	addr, err := btcutil.NewAddressPubKeyHash(addrHash, chainParams)
+	for i, test := range tests {
+		addrHash := test.in
+
+		var addr btcutil.Address
+		var err error
+		switch test.addrType.(type) {
+		case *btcutil.AddressPubKeyHash:
+			addr, err = btcutil.NewAddressPubKeyHash(addrHash, chainParams)
+		case *btcutil.AddressScriptHash:
+			addr, err = btcutil.NewAddressScriptHashFromHash(addrHash, chainParams)
+		}
+		if err != nil {
+			tc.t.Errorf("%s #%d: NewAddress unexpected error: %v", prefix, i, err)
```

Comment:
Should be fatal, or put a continue here so it doesn't keep trying to use the invalid address.